dspy/predict/predict.py

```diff
@@ -16,6 +18,35 @@ def warn_once(msg: str):
     logging.warning(msg)
 
 
+#-------------------------------------------------------------------------------
+#    Helper functions
+#-------------------------------------------------------------------------------
+
+def assert_structural_equivalency_for_predictors(
+        predictor1: object,
+        predictor2: object,
+    ) -> Optional[AssertionError]:
```

Comment:
Bad output type? This function returns None.

Reply:
Returns an `AssertionError` if the assert statement fails -- type hint is accounting for that. Return type is optional `AssertionError`, because it returns `None` if the assert statements succeed.

Reply:
It doesn't return it throws. Im in favor of leaving the type off this one

Reply:
Yes this type hint is a bit confusing, we can omit the type hints here.